### PR TITLE
#313: Add Camera / ProjectionMatrix to ShapeRenderer

### PR DESCRIPTION
**Pull request by deveth0**

### CODE REVIEW

File: graphics/src/main/kotlin/ktx/graphics/shapeRenderer.kt [hunk 1]
```diff
@@ -167,12 +169,30 @@ fun ShapeRenderer.triangle(
...
+
+/**
+ * Automatically calls [ShapeRenderer.begin] and [ShapeRenderer.end].
+ * @param type specified shape type used to draw the shapes in the [action] block. Can be changed during the rendering
+ * with [ShapeRenderer.set].
+ * @param camera The camera's [Camera.combined] matrix will be set to the ShapeRenderer's projection matrix before [ShapeRenderer.begin]
+ * @param action inlined. Executed after [ShapeRenderer.begin] and before [ShapeRenderer.end].
+ */
+@OptIn(ExperimentalContracts::class)
+inline fun <SR : ShapeRenderer> SR.use(type: ShapeType, camera: Camera, action: (SR) -> Unit) {
```
Comment (czyzby) [inline graphics/src/main/kotlin/ktx/graphics/shapeRenderer.kt:195]: Can you also add a test for this new method?

**Comment by czyzby** [comment]
Looking good so far, this addition makes sense.

File: graphics/src/main/kotlin/ktx/graphics/shapeRenderer.kt (hunk 1 above)
Comment (deveth0) [reply graphics/src/main/kotlin/ktx/graphics/shapeRenderer.kt:195]: sure, done :)

**Comment by czyzby** [comment]
Thanks for your contribution.

**Comment by deveth0** [comment]
Thanks for handling this so fast :)

**Pull request merged**